Venmo/src/main/java/com/braintreepayments/api/venmo/VenmoLauncher.kt

```diff
@@ -7,15 +7,23 @@ import com.braintreepayments.api.BrowserSwitchClient
 import com.braintreepayments.api.BrowserSwitchException
 import com.braintreepayments.api.BrowserSwitchFinalResult
 import com.braintreepayments.api.BrowserSwitchStartResult
+import com.braintreepayments.api.core.AnalyticsClient
 import com.braintreepayments.api.core.BraintreeException
 
 /**
  * Responsible for launching the Venmo app to authenticate users
  */
 class VenmoLauncher internal constructor(
-    private val browserSwitchClient: BrowserSwitchClient
+    private val browserSwitchClient: BrowserSwitchClient,
+    lazyAnalyticsClient: Lazy<AnalyticsClient>
```

Comment:
I ran into a crash where the SdkComponent was not yet initialized when the `VenmoLauncher` was created before the `VenmoClient`. This is because the creation of the `VenmoClient` initializes the `BraintreeClient` which creates the SdkComponent instance.

We'll need to remain flexible on the ordering of these classes.

The main issue is that `Context` is not passed into the SDK until the Client class is created. So any dependencies that reference context, such as these Launcher classes, would need to have those dependencies marked as lazy in this approach.

I think long term it may become confusing on which dependencies need to be lazy which is why I'm on the fence on this solution. But I wasn't able to find a cleaner solution that fixes this DI issue.